src/metal.zig

```diff
@@ -251,32 +262,118 @@ pub const Device = struct {
     }
 };
 
+pub const StreamingManager = struct {
+    const InflightBuffer = struct {
+        mtl_buffer: *mtl.Buffer,
+        queue: *Queue,
+        fence_value: u64,
+    };
+
+    mtl_device: *mtl.Device,
+    inflight_buffers: std.ArrayList(InflightBuffer),
+    free_buffers: std.ArrayList(*mtl.Buffer),
+
+    pub fn init(mtl_device: *mtl.Device) !*StreamingManager {
+        var manager = try allocator.create(StreamingManager);
+        manager.* = .{
+            .mtl_device = mtl_device,
+            .inflight_buffers = std.ArrayList(InflightBuffer).init(allocator),
+            .free_buffers = std.ArrayList(*mtl.Buffer).init(allocator),
+        };
+        return manager;
+    }
+
+    pub fn deinit(manager: *StreamingManager) void {
+        for (manager.inflight_buffers.items) |inflight_buffer| {
+            inflight_buffer.mtl_buffer.release();
+        }
+        for (manager.free_buffers.items) |mtl_buffer| {
+            mtl_buffer.release();
+        }
```

Comment:
nit: single-line for loops like this can be written inline unless the line is super lengthy

```zig
for (manager.free_buffers.items) |mtl_buffer| mtl_buffer.release();
```

Reply:
Sounds good.  Maybe add something to https://machengine.org/about/style/ for when you want to omit braces and when to go on a single-line?